core/src/workflows/application.rs

```diff
@@ -26,7 +35,18 @@ pub async fn initialize(
         context.log("dna/initialize: Initializing new chain from given DNA...");
         await!(initialize_chain(dna.clone(), &instance_context))?;
     }
+
+    // 3. Initialize the network
     await!(initialize_network(&instance_context))?;
+
+    // 4. Call publish on the agent and DNA entries. 
+    // This is to trigger the publishing of their headers not the entries themselves
```

Comment:
Ah!
I guess this is what breaks hachiko's assumptions and makes it wait for the DNA entry to be held by everybody since it sees the publish action but then no subsequent hold!